dht.go

```diff
@@ -109,8 +112,8 @@ func New(ctx context.Context, h host.Host, options ...opts.Option) (*IpfsDHT, er
 // NewDHT creates a new DHT object with the given peer as the 'local' host.
 // IpfsDHT's initialized with this function will respond to DHT requests,
 // whereas IpfsDHT's initialized with NewDHTClient will not.
-func NewDHT(ctx context.Context, h host.Host, dstore ds.Batching) *IpfsDHT {
-	dht, err := New(ctx, h, opts.Datastore(dstore))
+func NewDHT(ctx context.Context, h host.Host, dstore ds.Batching, bucketSize int) *IpfsDHT {
```

Comment:
A better option would be to modify `dht-node` so it constructs using `New` instead of `NewDHT`, so that we don't have to expand the `NewDHT` interface, which I know we don't want to do. I'll make that change, but it shouldn't prevent someone from reviewing.